### PR TITLE
feature/E-CLASSROOM-109-API-user-profile-functionality

### PR DESCRIPTION
**Pull request by Teraaay**
### Issue Link

https://framgiaph.backlog.com/view/E_CLASSROOM-109

### Definition of Done
- [x] Create api routes for returning specific user details and editing user details
- [x] Create/modify 'UserController' and create these functionalities:
    - [x] View specific user data
    - [x] Update user data

### Commands to Run

N/A

#### Main note
- Since there is no seeder for user yet, I just inserted a sample data directly to my local database just to test my routes and functionality if it's working.
#### Pages Affected
- N/A

#### Scenarios/ Test Cases
- [x] Create api routes for returning specific user details and editing user details
- [x] Create/modify 'UserController' and create these functionalities:
    - [x] View specific user data
    - [x] Update user data

### Screenshots
Created a sample data 
image

Sample of Viewing Specific User
image

Sample of Editing User Info
image

### CODE REVIEW

**Comment by geneno99** [comment]
### Approved PR #4 

- [x] Able to GET response for user API
- [x] Able to GET response when user create and update user details

<details><summary>Screenshot</summary>

post1

post2
</details>

-----
- Total Test time: 0.5 hr
- Total Iterations: 0
- Total Comments: 0

**Comment by syntaxerron** [comment]
- [x] Create trait for the response message, both for error response and success response. (Place inside "app/Traits/" directory)
- [x] Place your created controllers inside "app/Http/Controllers/API/v1" directory
- [x] Group API routes with 'v1'
- [x] Create Form Request to separate your validation with controller for update() (Refer to this: https://laravel.com/docs/8.x/validation#form-request-validation). Place all form request to "app/Http/Requests/User" directory. (For your Form Request authorize() method, return true for now since no functionality for authorization yet).

File: routes/api.php [hunk 1]
```diff
@@ -18,4 +18,4 @@
     return $request->user();
 });
 
-Route::resource('user', UserController::class);
+Route::resource('v1/user', UserController::class);
```
Comment (erronerron) [inline routes/api.php:21]: please use this approach instead:
Route::prefix('v1')->group(function () {
    Route::resource('/user', UserController::class);
});

Comment (syntaxerron) [inline routes/api.php]: please use this approach instead:
Route::prefix('v1')->group(function () {
    Route::resource('/user', UserController::class);
});

**Comment by syntaxerron** [comment]
- [x] Place "UpdateUserRequest" inside folder "User" within app/Http/Requests

File: routes/api.php [hunk 2]
```diff
@@ -18,4 +18,6 @@
     return $request->user();
 });
 
-Route::resource('v1/user', UserController::class);
+Route::prefix('v1')->group(function () {
+    Route::resource('/user', UserController::class);
```
Comment (syntaxerron) [inline routes/api.php]: by convention, please use plural form on routes: '/users' instead of '/user'

**Pull request merged**